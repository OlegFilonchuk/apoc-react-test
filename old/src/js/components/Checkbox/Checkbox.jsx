import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Checkbox.less';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

export const labelPositions = {
  left: 'left',
  right: 'right'
};

let globalCheckboxIndex = 0;

/**
 * Checkbox Component
 */
export default class Checkbox extends React.Component {
  componentWillMount() {
    globalCheckboxIndex += 1;
    this.id = `checkbox-gid-${globalCheckboxIndex}`;
  }

  render() {
    const {
      label,
      labelPosition,
      fullWidth,
      disabled,
      name,
      isOnlyCheckboxClickable,
      showHint,
      className,
      tooltip,
      ...props
    } = this.props;

    const id = name || this.id;

    const classes = classNames('square-checkbox', className, {
      'square-checkbox--clickable': !isOnlyCheckboxClickable,
      'square-checkbox-full-width': fullWidth,
      'square-checkbox-disabled': disabled
    });

    const labelTitle = showHint ? tooltip : null;

    const squareElement = (
      <label className="square-checkbox-square">
        <input
          type="checkbox"
          {...props}
          id={id}
          name={name}
          disabled={disabled}
          ref={ref => {
            this.input = ref;
          }}
          {...getDataTestElementProps(this.props)}
        />
        <span />
      </label>
    );

    const forElementId = !isOnlyCheckboxClickable ? id : '';
    const labelClasses = classNames('square-checkbox__label', {
      'square-checkbox__label--clickable': !isOnlyCheckboxClickable
    });
    const labelElement = (
      <label htmlFor={forElementId} className={labelClasses} title={labelTitle}>
        {label}
      </label>
    );

    const isLabelLeftPosition = labelPosition === labelPositions.left;
    const [firstItem, secondItem] = isLabelLeftPosition ? [labelElement, squareElement] : [squareElement, labelElement];

    return (
      <div className={classes}>
        {firstItem}
        {secondItem}
      </div>
    );
  }
}

Checkbox.propTypes = {
  /**
   * An additional class name that can be added to checkbox label for
   * extending it's functionality
   */
  className: PropTypes.string,

  /**
   * If set to yes label and checkbox fits the opposite edges of its container,
   * otherwise are positioned like two inline elements next to each other
   */
  fullWidth: PropTypes.bool,

  /**
   * Determines whether label should be positioned left from checkbox or right
   */
  labelPosition: PropTypes.oneOf([labelPositions.left, labelPositions.right]),

  /**
   * A flag to show title hint on hover
   */
  showHint: PropTypes.bool,

  /**
   * Label text or element.
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,

  /**
   * This value will be shown as a hint on mouse over
   */
  tooltip: PropTypes.string,

  /**
   * Whether checkbox is disabled or not
   */
  disabled: PropTypes.bool,

  /**
   * It's passed to `input` element. Used also to create an `id` attribute of input
   * and `for` of label in order to connect those two.
   */
  name: PropTypes.string,

  /**
   * If is set to true label will not be clickable
   */
  isOnlyCheckboxClickable: PropTypes.bool,

  /**
   * Is the checkbox checked
   */
  checked: PropTypes.bool,
  ...dataTestElementPropTypes
};

Checkbox.defaultProps = {
  className: '',
  tooltip: '',
  name: '',
  disabled: false,
  fullWidth: false,
  showHint: false,
  isOnlyCheckboxClickable: false,
  labelPosition: labelPositions.right,
  checked: false
};
