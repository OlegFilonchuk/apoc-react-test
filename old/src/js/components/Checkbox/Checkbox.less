@disabledColor: #aeaeae;
@checkedColor: #555555;
@hoverColor: #aeaeae;

.square-checkbox {
  color: @checkedColor;
  display: flex;
  align-items: center;
  justify-content: flex-start;

  &__label {
    margin-bottom: 0;
    flex: 1 1;

    &--clickable {
      cursor: pointer;
    }
  }

  @squareSize: 12px;
  &-square {
    flex: 0 0 @squareSize;
    vertical-align: middle;
    display: inline-block;
    width: @squareSize;
    height: @squareSize;
    border: 1px solid @checkedColor;
    background-color: transparent;
    position: relative;
    margin: 0;

    input[type='checkbox'] {
      display: none;

      & + span {
        background-color: transparent;
        position: absolute;
        top: 1px;
        left: 1px;
        right: 1px;
        bottom: 1px;
        cursor: pointer;
        margin-bottom: 0;
      }
    }

    input[type='checkbox'][disabled]:not(:checked) + span {
      background-color: transparent;
    }

    input[type='checkbox'][disabled]:checked + span {
      background-color: @disabledColor;
    }

    input[type='checkbox']:checked:not([disabled]) + span {
      background-color: @checkedColor;
    }

    &:hover {
      input[type='checkbox']:not(:checked):not([disabled]) + span {
        background-color: @hoverColor;
      }
    }
  }

  &--clickable {
    &:hover {
      .square-checkbox-square {
        input[type='checkbox']:not(:checked):not([disabled]) + span {
          background-color: @hoverColor;
        }
      }
    }
  }

  &__label + &-square {
    margin-left: 8px;
  }

  &-square + &__label {
    margin-left: 8px;
  }
}

.square-checkbox.square-checkbox-disabled {
  color: @disabledColor;

  .square-checkbox-square {
    border-color: @disabledColor;
    cursor: not-allowed;

    > span {
      cursor: inherit;
    }
  }

  .square-checkbox__label.square-checkbox__label--clickable {
    cursor: not-allowed;
  }
}

.square-checkbox.square-checkbox-full-width {
  justify-content: space-between;

  .square-checkbox__label,
  .square-checkbox-square {
    margin-left: 0;
  }

  .square-checkbox-square:first-child + .square-checkbox__label,
  .square-checkbox__label:first-child + .square-checkbox-square {
    margin-left: auto;
  }
}
