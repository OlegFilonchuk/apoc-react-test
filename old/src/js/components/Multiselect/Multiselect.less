.multiselect {
  list-style: none;
  padding: 0;
  margin: 0;

  &--no-vertical-lines {
    li {
      padding-top: 2px;
      padding-bottom: 3px;

      &:last-child {
        padding-bottom: 2px;
      }
    }
  }

  &:not(&--no-vertical-lines) {
    li {
      border-bottom: 1px solid #e4e4e4;
      padding: 2px 0;

      &:last-child {
        border-bottom: 0;
      }
    }
  }

  li {
    font-size: 12px;

    .li-content {
      overflow: hidden;
      padding: 0 15px;
      color: #555555;
    }
  }
}
