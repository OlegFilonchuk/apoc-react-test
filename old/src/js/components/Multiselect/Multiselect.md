### Example

    function onMultiselectChange(filterState) {
      const multiSelectValue = Array.from(filterState).map( (nameAndIsChecked, index) => {
        const name = nameAndIsChecked[0];
        const isChecked = nameAndIsChecked[1];
        return (
          <span key={index}>
            {name} : {isChecked.toString()} : {name}
            <br/>
          </span>
        )
      });

      setState({multiSelectValue});
    }

    function onCheckboxChange(e) {
      setState({checkboxValue : `${e.target.name}-${e.target.checked}`});
    }

    <div>
      <p>Multiselect</p>
      <Multiselect
        onChange={onMultiselectChange}
      >
        <Checkbox
          label={'Why does the cloud whine?'}
          defaultChecked
          onChange={onCheckboxChange}
        />
        <Checkbox
          label={'Well, yer not fighting me without a life!'}
          defaultChecked={false}
        />
        <Checkbox
          label={'Where is the golden girl?'}
          defaultChecked
          disabled
        />
        <Checkbox
          label={'How coal-black. You crush like a cloud.'}
          defaultChecked={false}
          disabled
        />
      </Multiselect>
      <p>Multiselect onChange result:</p>
      <p style={{fontSize : '0.8em'}}>{state.multiSelectValue}</p>
      <p>First checkbox onChange result:</p>
      <p style={{fontSize : '0.8em'}}>{state.checkboxValue}</p>
    </div>


### Example with bi-directional data flow

    const initialState = {
      isFirstChecked: false,
      isSecondChecked: false,
      booleanValue: false
    }

    function onMultiselectChange(mapKeysValues){
      setState({
        isFirstChecked: mapKeysValues.get('first'),
        isSecondChecked: mapKeysValues.get('second')
      })
    }

    function toggleFirstCheckbox(){
      setState({
        isFirstChecked: !state.isFirstChecked,
      })
    }

    function toggleSecondCheckbox(){
      setState({
        isSecondChecked: !state.isSecondChecked,
      })
    }

    function toggleBooleanValue(){
      setState({
        booleanValue: !state.booleanValue,
      })
    }

    <div>
      <Multiselect onChange={onMultiselectChange}>
        <Checkbox
          label={'Why does the cloud whine?'}
          name="first"
          checked={state.isFirstChecked}
        />
        <Checkbox
          label={'Well, yer not fighting me without a life!'}
          name="second"
          checked={state.isSecondChecked}
        />
       </Multiselect>
       <p>
         Is first checked: {String(state.isFirstChecked)}
       </p>
       <p>
         Is second checked: {String(state.isSecondChecked)}
       </p>
       <button onClick={toggleFirstCheckbox}>Toggle first checkbox</button>
       <button onClick={toggleSecondCheckbox}>Toggle second checkbox</button>
    </div>
    
### Example without horizontal lines between checkboxes

 

    <Multiselect onChange={()=>null} noVerticalLines>
      <Checkbox
        label={'Why does the cloud whine?'}
        name="first"
        checked={false}
      />
      <Checkbox
        label={'Well, yer not fighting me without a life!'}
        name="second"
        checked={true}
      />
      <Checkbox
        label={'The klingon resists nuclear flux like an apocalyptic particle!'}
        name="second"
        checked={true}
      />
     </Multiselect>       
