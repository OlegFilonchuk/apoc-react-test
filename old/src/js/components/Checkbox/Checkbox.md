## Chameleon input\[type="checkbox"\]
This is example of checkbox which is just static html and css from Chameleon.

    <div className="sc-checkbox">
      <label>
        <input type="checkbox"/>
        Check me out
      </label>
    </div>

## Checkbox
Following are examples of Checkbox react component, which is in fact
a wrapper of html element:

```html
<input type="checkbox"/>
```

with no changes in functionality comparing to original `input`.
All non-Checkbox specific props passed to `Checkbox` are passed to `input`.

### Label position

By default, label position is to the right

    <div>
      <Checkbox label="Lorem ipsum"/>
    </div>

### Label position to the left
Position can be changed using `labelPosition` prop. All possible
values are stored in `labelPositions` map. It can be imported in following
way:

```html
import Checkbox, {labelPositions} from 'apoc-react/Checkbox';

```

    const labelPositions = require('./Checkbox').labelPositions;

    <Checkbox label="Lorem ipsum" labelPosition={labelPositions.left}/>

### Disabled

    <div>
      <Checkbox label="Lorem ipsum" disabled/>
    </div>

### Full width

    <div>
      <Checkbox label="Lorem ipsum" fullWidth/>
    </div>

### Checked by default

    <div>
      <Checkbox label="Lorem ipsum" defaultChecked/>
    </div>

### Passing additional attributes
Specific to `input` element

    const onCheckboxChange = e => {
      setState({
        checkboxValue: `${e.target.name}:${e.target.value}`,
        checkboxChecked: `${e.target.checked}`
      });
    };

    (
      <div>
        <p>Current value is: {state.checkboxValue}</p>
        <p>Is checked: {state.checkboxChecked}</p>
        <Checkbox name={'foo'} onChange={onCheckboxChange} label={'Click me!'} value="IamSelected" />
      </div>
    )

### Passing element as a title
The `title` prop can accept element instead of string,

    <div>
      <Checkbox
        label={(
          <span>
            <b>Strong title</b>{'\u00A0'}
            <i>italic part</i>{'\u00A0'}
            <u>part with underline</u>
          </span>
        )}
      />
    </div>

### Bi-directional flow of checked state. Changing state from the outside of Checkbox component.
The `title` prop can accept element instead of string,

    const initialState = { isChecked : false }

    function toggleCheckbox(){
      setState({isChecked: !state.isChecked});
    };

    function onCheckboxChange(e){
      setState({isChecked: e.target.checked});
    };

    (
      <div>
        <Checkbox
          label="This is checkbox"
          checked={state.isChecked}
          onChange={onCheckboxChange}
        />
        <br/>
        <p><span>Is checked?: {String(state.isChecked)}</span></p>
        <br/>
        <button onClick={toggleCheckbox}>Toggle checkbox</button>
      </div>
    )

### With title hint
    <div>
      <Checkbox label="Lorem ipsum" showHint={true} tooltip="some example hint" />
    </div>

### Only checkbox clickable example

By default, label position is to the right

    <div>
      <Checkbox label="Lorem ipsum" isOnlyCheckboxClickable/>
    </div>
