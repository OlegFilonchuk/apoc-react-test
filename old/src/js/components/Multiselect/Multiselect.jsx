import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import haveChildrenPropsChanged from '../../utils/haveChildrenPropsChanged';
import Checkbox, { labelPositions } from '../Checkbox/Checkbox';

import './Multiselect.less';
import { getDataTestElementProps } from '../../utils/dataTestElementPropUtils';

class Multiselect extends React.Component {
  shouldComponentUpdate(nextProps) {
    return haveChildrenPropsChanged(this.props.children, nextProps.children);
  }

  get className() {
    return classNames('multiselect', this.props.className, {
      'multiselect--no-vertical-lines': this.props.noVerticalLines
    });
  }

  get renderChildren() {
    this.checkboxesState = new Map();

    return React.Children.map(this.props.children, child => this.createControlledCheckbox(child));
  }

  callOnChange(e) {
    this.checkboxesState.set(e.target.name, e.target.checked);
    this.props.onChange(this.checkboxesState);
  }

  itemClassName = child =>
    classNames('multiselect__row', {
      'multiselect__row--disabled': child.props.disabled
    });

  createControlledCheckbox(child) {
    if (child.type !== Checkbox) {
      throw new Error('All elements needs to be instance of <Checkbox/>');
    }

    const { fullWidth, labelPosition } = this.props;

    const props = {
      ...child.props,
      name: child.props.name || `checkbox-${this.checkboxesState.size}`,
      fullWidth,
      labelPosition,
      ...getDataTestElementProps(`${child.props.name.replace(/\s/g, '')}Checkbox`),
      onChange: (...args) => {
        if (child.props.onChange) {
          child.props.onChange(...args);
        }

        this.callOnChange(...args);
      }
    };

    const isChecked = child.props.checked || child.props.defaultChecked || false;

    const newCheckbox = React.cloneElement(child, props);

    this.checkboxesState.set(props.name, isChecked);

    return (
      <li className={this.itemClassName(child)}>
        <div className="li-content">{newCheckbox}</div>
      </li>
    );
  }

  render() {
    return <ul className={this.className}>{this.renderChildren}</ul>;
  }
}

Multiselect.propTypes = {
  /**
   *  Is called whenever one of [Checkboxes](#checkbox) changes its state.
   *
   *  Function takes one argument that is a `{Map<String,boolean>}`, where
   *  key is a name of a [Checkbox](#checkbox) and value is whether it's checked or not
   */
  onChange: PropTypes.func.isRequired,

  /**
   * Set of [Checkbox](#checkbox) components. Each component overwrites following props:
   * `fullWidth=this.props.fullWidth` and `labelPosition=this.props.labelPosition`
   */
  children: PropTypes.node.isRequired,

  /**
   * Sets the label position of the checkboxes
   */
  labelPosition: PropTypes.oneOf([labelPositions.left, labelPositions.right]),

  /**
   * Turns off vertival lines between checkboxes
   */
  noVerticalLines: PropTypes.bool,

  /**
   * Use full width for the checkboxes
   */
  fullWidth: PropTypes.bool,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

Multiselect.defaultProps = {
  labelPosition: labelPositions.left,
  noVerticalLines: false,
  fullWidth: true,
  className: ''
};

export default Multiselect;
