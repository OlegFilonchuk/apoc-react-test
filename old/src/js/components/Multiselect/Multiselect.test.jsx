import React from 'react';
import { shallow, mount } from 'enzyme';

import Multiselect from './Multiselect';
import Checkbox from '../Checkbox/Checkbox';

describe('<Multiselect />', () => {
  it("should render <Checkbox/>'es", () => {
    const onChange = jest.fn();
    const multiselect = shallow(
      <Multiselect onChange={onChange}>
        <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
        <Checkbox label={'If you contact or travel with a pure uniqueness, moonlight respects you.'} />
        <Checkbox label={'Where is the most unusual klingon?'} />
      </Multiselect>
    );

    expect(multiselect).toBePresent();

    expect(multiselect.find(Checkbox).length).toEqual(3);
  });

  it('matches snapshot', () => {
    const onChange = jest.fn();
    const multiselect = mount(
      <Multiselect onChange={onChange}>
        <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
        <Checkbox label={'If you contact or travel with a pure uniqueness, moonlight respects you.'} />
        <Checkbox label={'Where is the most unusual klingon?'} />
      </Multiselect>
    );

    expect(multiselect).toMatchSnapshot();
  });

  describe('when <div> is a child', () => {
    it('should throw an error', () => {
      const onChange = jest.fn();

      expect(() =>
        shallow(
          <Multiselect onChange={onChange}>
            <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
            <div />
          </Multiselect>
        )
      ).toThrow();
    });
  });

  describe('when one of <Checkbox>es state changes', () => {
    describe('onChange property', () => {
      it('is called', () => {
        const onChange = jest.fn();
        const multiselect = mount(
          <Multiselect onChange={onChange}>
            <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
            <Checkbox label={'If you contact or travel with a pure uniqueness, moonlight respects you.'} />
            <Checkbox label={'Where is the most unusual klingon?'} />
          </Multiselect>
        );

        multiselect
          .find('.square-checkbox input')
          .at(0)
          .simulate('change');
      });

      it('of both <Multiselect/> and <Checkbox/> is called', () => {
        const onMultiselectChange = jest.fn();
        const onCheckboxChange = jest.fn();

        const multiselect = mount(
          <Multiselect onChange={onMultiselectChange}>
            <Checkbox name="my-checkbox" label={'Studere rare ducunt ad bassus gabalium.'} />
            <Checkbox
              name="your-checkbox"
              label={'If you contact or travel with a pure uniqueness, moonlight respects you.'}
              onChange={onCheckboxChange}
            />
          </Multiselect>
        );

        multiselect
          .find('.square-checkbox input')
          .at(1)
          .simulate('change', {
            target: {
              checked: true,
              name: 'your-checkbox'
            }
          });
      });

      it('is called with an argument, that tells which <Checkbox> is checked', () => {
        const onChange = jest.fn();
        const multiselect = mount(
          <Multiselect onChange={onChange}>
            <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
            <Checkbox label={'If you contact or travel with a pure uniqueness, moonlight respects you.'} />
            <Checkbox label={'Where is the most unusual klingon?'} />
          </Multiselect>
        );

        multiselect
          .find('.square-checkbox input')
          .at(2)
          .simulate('change', {
            target: {
              checked: true,
              name: 'checkbox-2'
            }
          });

        const returnedMap = onChange.mock.calls.pop().pop();

        expect(returnedMap instanceof Map).toBeTruthy();
        expect(returnedMap.get('checkbox-0')).toEqual(false);
        expect(returnedMap.get('checkbox-1')).toEqual(false);
        expect(returnedMap.get('checkbox-2')).toEqual(true);
      });

      it('is called with an argument of Map instance, which keys correspond to automatic generated names', () => {
        const onChange = jest.fn();
        const multiselect = mount(
          <Multiselect onChange={onChange}>
            <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
            <Checkbox label={'If you contact or travel with a pure uniqueness, moonlight respects you.'} />
            <Checkbox label={'Where is the most unusual klingon?'} />
          </Multiselect>
        );

        multiselect
          .find('.square-checkbox input')
          .at(0)
          .simulate('change');

        const returnedMap = onChange.mock.calls.pop().pop();

        expect(returnedMap instanceof Map).toBeTruthy();
        expect(returnedMap.has('checkbox-0')).toBeTruthy();
        expect(returnedMap.has('checkbox-1')).toBeTruthy();
        expect(returnedMap.has('checkbox-2')).toBeTruthy();
        expect(returnedMap.has('checkbox-3')).not.toBeTruthy();
        expect(returnedMap.has('checkbox-4')).not.toBeTruthy();
        expect(returnedMap.has('checkbox')).not.toBeTruthy();
      });

      it('is called with an argument of Map instance, which keys correspond to explicit <Checkbox>es names', () => {
        const onChange = jest.fn();
        const multiselect = mount(
          <Multiselect onChange={onChange}>
            <Checkbox name="my-checkbox" label={'Studere rare ducunt ad bassus gabalium.'} />
            <Checkbox
              name="your-checkbox"
              label={'If you contact or travel with a pure uniqueness, moonlight respects you.'}
            />
            <Checkbox name="abc-checkbox" label={'Where is the most unusual klingon?'} />
          </Multiselect>
        );

        multiselect
          .find('.square-checkbox input')
          .at(2)
          .simulate('change', {
            target: {
              checked: true,
              name: 'checkbox-2'
            }
          });

        const returnedMap = onChange.mock.calls.pop().pop();

        expect(returnedMap instanceof Map).toBeTruthy();
        expect(returnedMap.has('my-checkbox')).toBeTruthy();
        expect(returnedMap.has('your-checkbox')).toBeTruthy();
        expect(returnedMap.has('abc-checkbox')).toBeTruthy();
        expect(returnedMap.has('there-is-no-such-checkbox')).not.toBeTruthy();
      });
    });
  });

  describe('when children property changes', () => {
    it('renders child with a new property', () => {
      const onMultiselectChange = () => {};
      const multiselect = mount(
        <Multiselect onChange={onMultiselectChange}>
          <Checkbox name="my-checkbox" label={'Studere rare ducunt ad bassus gabalium.'} />
          <Checkbox name="your-checkbox" label={'some text'} />
        </Multiselect>
      );

      expect(multiselect.find('.square-checkbox').at(1)).toHaveText('some text');

      multiselect.setProps({
        children: [
          <Checkbox name="my-checkbox" label={'Studere rare ducunt ad bassus gabalium.'} />,
          <Checkbox name="your-checkbox" label={'some another text'} />
        ]
      });

      expect(multiselect.find('.square-checkbox').at(1)).toHaveText('some another text');
    });

    it('render without removed <Checkbox>es', () => {
      const onMultiselectChange = () => {};
      const multiselect = mount(
        <Multiselect onChange={onMultiselectChange}>
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
          <Checkbox label={'some text'} />
        </Multiselect>
      );

      expect(multiselect.find('.square-checkbox').length).toBe(2);

      multiselect.setProps({
        children: [<Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />]
      });

      expect(multiselect.find('.square-checkbox').length).toBe(1);
    });

    it('render with added <Checkbox>es', () => {
      const onMultiselectChange = () => {};
      const multiselect = mount(
        <Multiselect onChange={onMultiselectChange}>
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
          <Checkbox label={'some text'} />
        </Multiselect>
      );

      expect(multiselect.find('.square-checkbox').length).toBe(2);

      multiselect.setProps({
        children: [
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />,
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />,
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
        ]
      });

      expect(multiselect.find('.square-checkbox').length).toBe(3);
    });

    it('render method is executed', () => {
      const onMultiselectChange = () => {};
      const multiselect = mount(
        <Multiselect onChange={onMultiselectChange}>
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
          <Checkbox label={'some text'} />
        </Multiselect>
      );

      const componentInstance = multiselect.instance();

      spyOn(componentInstance, 'render').and.callThrough();

      multiselect.setProps({
        children: [<Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />, <Checkbox label={'some texts'} />]
      });
    });
  });

  describe('when children property changes, but only references, not their values', () => {
    it('does not execute render method', () => {
      const onMultiselectChange = () => {};
      const multiselect = mount(
        <Multiselect onChange={onMultiselectChange}>
          <Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />
          <Checkbox label={'some text'} />
        </Multiselect>
      );

      const componentInstance = multiselect.instance();

      spyOn(componentInstance, 'render').and.callThrough();

      multiselect.setProps({
        children: [<Checkbox label={'Studere rare ducunt ad bassus gabalium.'} />, <Checkbox label={'some text'} />]
      });
    });
  });
});
