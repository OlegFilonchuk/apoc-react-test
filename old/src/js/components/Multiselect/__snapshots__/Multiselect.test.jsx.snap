// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Multiselect /> matches snapshot 1`] = `
<Multiselect
  className=""
  fullWidth={true}
  labelPosition="left"
  noVerticalLines={false}
  onChange={[Function]}
>
  <ul
    className="multiselect"
  >
    <li
      className="multiselect__row"
      key=".0"
    >
      <div
        className="li-content"
      >
        <Checkbox
          checked={false}
          className=""
          data-test-element="Checkbox"
          disabled={false}
          fullWidth={true}
          isOnlyCheckboxClickable={false}
          label="Studere rare ducunt ad bassus gabalium."
          labelPosition="left"
          name="checkbox-0"
          onChange={[Function]}
          showHint={false}
          tooltip=""
        >
          <div
            className="square-checkbox square-checkbox--clickable square-checkbox-full-width"
          >
            <label
              className="square-checkbox__label square-checkbox__label--clickable"
              htmlFor="checkbox-0"
              title={null}
            >
              Studere rare ducunt ad bassus gabalium.
            </label>
            <label
              className="square-checkbox-square"
            >
              <input
                checked={false}
                data-test-element="Checkbox"
                disabled={false}
                id="checkbox-0"
                name="checkbox-0"
                onChange={[Function]}
                type="checkbox"
              />
              <span />
            </label>
          </div>
        </Checkbox>
      </div>
    </li>
    <li
      className="multiselect__row"
      key=".1"
    >
      <div
        className="li-content"
      >
        <Checkbox
          checked={false}
          className=""
          data-test-element="Checkbox"
          disabled={false}
          fullWidth={true}
          isOnlyCheckboxClickable={false}
          label="If you contact or travel with a pure uniqueness, moonlight respects you."
          labelPosition="left"
          name="checkbox-1"
          onChange={[Function]}
          showHint={false}
          tooltip=""
        >
          <div
            className="square-checkbox square-checkbox--clickable square-checkbox-full-width"
          >
            <label
              className="square-checkbox__label square-checkbox__label--clickable"
              htmlFor="checkbox-1"
              title={null}
            >
              If you contact or travel with a pure uniqueness, moonlight respects you.
            </label>
            <label
              className="square-checkbox-square"
            >
              <input
                checked={false}
                data-test-element="Checkbox"
                disabled={false}
                id="checkbox-1"
                name="checkbox-1"
                onChange={[Function]}
                type="checkbox"
              />
              <span />
            </label>
          </div>
        </Checkbox>
      </div>
    </li>
    <li
      className="multiselect__row"
      key=".2"
    >
      <div
        className="li-content"
      >
        <Checkbox
          checked={false}
          className=""
          data-test-element="Checkbox"
          disabled={false}
          fullWidth={true}
          isOnlyCheckboxClickable={false}
          label="Where is the most unusual klingon?"
          labelPosition="left"
          name="checkbox-2"
          onChange={[Function]}
          showHint={false}
          tooltip=""
        >
          <div
            className="square-checkbox square-checkbox--clickable square-checkbox-full-width"
          >
            <label
              className="square-checkbox__label square-checkbox__label--clickable"
              htmlFor="checkbox-2"
              title={null}
            >
              Where is the most unusual klingon?
            </label>
            <label
              className="square-checkbox-square"
            >
              <input
                checked={false}
                data-test-element="Checkbox"
                disabled={false}
                id="checkbox-2"
                name="checkbox-2"
                onChange={[Function]}
                type="checkbox"
              />
              <span />
            </label>
          </div>
        </Checkbox>
      </div>
    </li>
  </ul>
</Multiselect>
`;
